### ISSUE TITLE
Create PurchaseManager and pull relevant methods from RCPurchases

### ISSUE THREAD

**Issue by beylmk** [RevenueCat/purchases-ios#685]
See notes in RCPurchases.m

blocked by: 
- [ ] #581

**Comment by beylmk** [comment]
closing, is duplicate of: https://github.com/RevenueCat/purchases-ios/issues/684
